lib/DxcSupport/HLSLOptions.cpp

```diff
@@ -1383,9 +1383,10 @@ int ReadDxcOpts(const OptTable *optionTable, unsigned flagsToInclude,
   return 0;
 }
 
-/// Sets up the specified DxcDllSupport instance as per the given options.
-int SetupDxcDllSupport(const DxcOpts &opts, dxc::DxcDllSupport &dxcSupport,
-                       llvm::raw_ostream &errors) {
+/// Sets up the specified SpecificDllLoader instance as per the given options.
+int SetupSpecificDllLoader(const DxcOpts &opts,
+                           dxc::SpecificDllLoader &dxcSupport,
+                           llvm::raw_ostream &errors) {
```

Comment:
I think this rename is kind of confusing - the type of the object isn't what's interesting here, we're actually modifying how our `dxcompiler.dll` is being loaded in particular. In any case, this should probably use an `IDxcLoader` as I've suggested overall.

Reply:
Yes, in some cases this should be more abstracted. However, in cases like `ShaderOpTest.cpp`, where an assignment is being made directly to a member variable that has been defined to be a SpecificDllLoader, I think it makes sense to have the set up function be accurate, and `SetSpecificDllLoader`

Reply:
I don't understand this comment - this function is never called in ShaderOpTest.cpp - it's only used in `dxc.cpp`, `dxr.cpp`, and the test in `dxc_batch.cpp`. All of these are dealing with dxcompiler.dll in particular and not just some arbitrary "specific dll"